### PR TITLE
✨ draft-pr: add options

### PR DESCRIPTION
**Pull request by lucasrodes**
## Add prefix options

- `--category` (or `-c`): PR category based on [our etiquette](https://www.notion.so/owid/Git-GitHub-etiquette-68e824bed1a747e1a01f2b771c695bf7). Will be used to add an emoji to the start of the PR title.
- `--scope` (or `-s`): brief summary of the PR scope (e.g. 'wizard')

### Examples

`etl d draft-pr <branch-name> -c data -t "Update VDEM data"` →  **:bar_chart: Update VDEM data**

`etl d draft-pr <branch-name> -c data -s vdem -t "update"` →  **:bar_chart: vdem: update**

`etl d draft-pr <branch-name> "Update VDEM data"` →  **Update VDEM data**

`etl d draft-pr <branch-name> -c enhance -s vdem` →  **🚧 Draft PR for branch <branch-name>** (i.e. no title is given, will ignore prefixes and use default title)

### Notes
- This brings us closer to harmonizing PR titles. It also encourages us to define the scope of our work beforehand. Still, one can still change the PR name later from GitHub, of course.
- I can see how this command could be made shorter in the future if we need to:
  - `etl d draft-pr` → `etl pr`
  - `-c data -s vdem -t "update"` → `-d vdem "update"`, where first is flag with PR category, and then two positional (free-text) arguments: scope, and title.
  - so it'd be `etl d draft-pr -c data -s vdem -t update` → `etl pr -d vdem update`

### Other modifications
In addition, I've added shorter argument names `-t` for `--title` and `-b` for `--base-branch`.

### CODE REVIEW

File: apps/utils/draft_pull_request.py [hunk 1]
```diff
@@ -27,6 +27,62 @@
...
+        "description": "work in progress - intermediate commits that will be explained later on",
+    },
+    "tests": {
+        "emoji": "✅",
+        "emoji_raw": ":white_check_mark:",
+        "description": "adding missing tests, refactoring tests, etc. No production code change",
+    },
+}
+description = "- " + "\n- ".join(
+    f"**{choice}: {choice_params['description']}" for choice, choice_params in PR_CATEGORIES.items()
```
Comment (pabloarosado) [inline apps/utils/draft_pull_request.py:84]: Not sure why you added the "**". I think they are not necessary.

File: apps/utils/draft_pull_request.py [hunk 2]
```diff
@@ -37,10 +93,27 @@
...
 @click.option(
-    "--base-branch", type=str, default=None, help="Name of base branch (if not given, 'master' will be used)."
+    "--base-branch", "-b", type=str, default=None, help="Name of base branch (if not given, 'master' will be used)."
+)
+@click.option("--title", "-t", type=str, default=None, help="Title of the PR and the first commit.")
+@click.option(
+    "--category",
+    "-c",
+    type=click.Choice(list(PR_CATEGORIES.keys()), case_sensitive=False),
+    help=f"Category of the PR. Only works if --title is used (it will be added to the PR title). A corresponding emoji will be pre-fixed to the PR title.\n {description}",
```
Comment (pabloarosado) [inline apps/utils/draft_pull_request.py:103]: Maybe rephrase it a bit: "Category of the PR (only relevant if --title is given). A corresponding emoji will be prepended to the title.\n {description}"

File: apps/utils/draft_pull_request.py [hunk 3]
```diff
@@ -37,10 +93,27 @@
...
 )
-@click.option("--title", type=str, default=None, help="Title of the PR and the first commit.")
-def cli(new_branch: Optional[str] = None, base_branch: Optional[str] = None, title: Optional[str] = None) -> None:
+@click.option(
+    "--scope",
+    "-s",
+    help="Scope of the PR. Only works if --title is used (it will be added to the PR title).\n\n\n**Examples**: 'demography' for data work on this field, 'etl.db' if working on specific modules, 'wizard', etc.",
```
Comment (pabloarosado) [inline apps/utils/draft_pull_request.py:108]: I don't think having a scope is very useful, the title itself is usually enough. But up to you.

**Review by pabloarosado — APPROVED** [review]
I was thinking of doing this! :-) Thanks for adding it! I'd prefer to simplify and not have a scope, but if you have a strong opinion in favour of having it, go ahead.

**Comment by lucasrodes** [comment]
you can still use the tool without a scope, i agree we shouldn't force users to use it... at least for now!
